### PR TITLE
feat: add ArtworksRailHomeViewSection tracking

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-1255] 

### Description

This PR adds legacy tracking events to artworks' rail sections.

https://github.com/user-attachments/assets/732d4a8b-4131-460c-89a8-d19cb7799e74

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1255]: https://artsyproduct.atlassian.net/browse/ONYX-1255?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 6774b16327df2f538940cad2ec4950c1fd83069f
</p>

File: src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx [hunk 1]
```diff
@@ -17,11 +20,24 @@ export const ArtworksRailHomeViewSection: React.FC<ArtworksRailHomeViewSectionPr
   const data = useFragment(fragment, section)
   const title = data.component?.title
   const artworks = extractNodes(data.artworksConnection)
+  const tracking = useTracking()
   const componentHref = "" // TODO: should be in schema
 
-  if (!artworks || artworks.length === 0) return null
+  if (!data.artworksConnection?.totalCount) {
```
Comment (dblandin) [inline src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx:26]: **suggestion**: If we're already requesting artwork data then I think we might as well use the presence of those objects for the conditional here. I don't think we gain much from switching to `totalCount` which requests an additional field and triggers some additional backend querying to provide.

File: src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx [hunk 2]
```diff
@@ -17,11 +20,24 @@ export const ArtworksRailHomeViewSection: React.FC<ArtworksRailHomeViewSectionPr
...
+    return null
+  }
+
+  const handleOnArtworkPress = (artwork: any, position: number) => {
+    tracking.trackEvent(
+      LegacyHomeAnalytics.artworkThumbnailTapEvent(
```
Comment (dblandin) [inline src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx:32]: **praise**: Clear naming here 👍

**Review by dblandin — APPROVED** [review]

File: src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx (hunk 1 above)
Comment (MounirDhahri) [reply src/app/Scenes/HomeView/Sections/ArtworksRailHomeViewSection.tsx:26]: cool - good thought. I will adjust

**Pull request merged**